code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/TokenUtils.java

```diff
@@ -95,6 +104,17 @@ public Account getAccountFromJWT() {
         return jsonData;
     }
 
+    private String retrieveJwt(String jwtKey) {
+        AuthToken authToken= authTokenDao.getTokenById(jwtKey);
+        if (authToken == null) {
+            return null;
+        }
+        if (!authToken.getProvider().equalsIgnoreCase(SecurityConstants.AUTH_PROVIDER.get())){
+            throw new ClientVisibleException(ResponseCodes.UNAUTHORIZED);
```

Comment:
If you just throw unauthorized, won't the user get stuck in a loop?  I think you should just ignore the token and treat it as though you didn't find it.